Assets/Scripts/Interaction/Interactor.cs

```diff
@@ -16,10 +16,16 @@ public class Interactor : MonoBehaviour
         private void Start()
         {
             Logger.log("Initializing Interactor: " + gameObject.name);
+
             if (itemDisplayText == null || itemDisplaySubtext == null)
             {
                 Logger.log("ItemDisplayText or ItemDisplaySubText is null!");
             }
+
+            if (origin == null)
+            {
+                origin = GetComponentInChildren<Camera>().transform;
```

Comment:
```suggestion
                origin = GetComponentInChildren<Camera>().transform;
                Logger.logWarning("Interactor origin transform is missing and has been automatically assigned. Please assign the transform manually");
```

`GetComponent<>()` should not be used, if avoidable. It is really unperformant. Warning the programmer to assign the object to resolve the problem one for all is a better practice here, I think.

Reply:
This is the Start() function so performance doesn’t really matter here. It‘s a stylistic choice: either connect the components in the inspector or edit them in code.
I went with a middle ground approach here to get the best from both worlds. Use the assigned component from the editor and when it‘s not set query it from the current gameObject.

I‘m open to changes though.